docs/articles/Vector-Indexes.md

```diff
@@ -0,0 +1,319 @@
+
+### Types of Indexing
+Let’s discuss some popular indexes involved in vector indexing.
+
+#### Flat Indexing
+Flat indexing is the most basic type of indexing, as it stores vectors in the database as they are. No special operation is performed for modification or optimization. When a query vector arrives, it is compared with every other vector within the database to generate a similarity score. Due to this, flat indexing is also considered a brute-force approach. Once the similarity scores are calculated, the top k closest matches are retrieved.
+
+Flat Indexing is a very precise algorithm, as it retrieves vectors with great accuracy. However, it is computationally expensive and not ideal in cases where the database consists of millions of records or more.
+
+#### Locality Sensitive Hashing (LSH)
+LSH optimizes the vector search operation by dividing the database elements into buckets. Here, we first use a hashing function to compute hashes for each vector element. Then, based on the hash’s similarity, the vectors are grouped together into buckets. Intuitively, each bucket contains a matching vector.
+Now, when a query vector appears, it is first hashed using the same function. Then, based on a hash, it is assigned a bucket containing all its similar vectors. The query vector now only needs to be compared with the bucket vectors, reducing the search space and dramatically improving the efficiency.
+
+#### Inverted File (IVF)
+IVF works similarly to LSH and creates groups of data. But rather than hashing it, it uses clustering techniques. The techniques can vary depending on the implementation, but simpler techniques may use K-means clustering and then use the cluster centroids as a reference for query vectors. The query vector is then compared with only its associated data cluster to improve efficiency. IVF has a few variations, each improving the storage and retrieval efficiency.
+
+##### IVFFLAT
+The flat variation clusters the vectors and creates centroids but stores each vector as it is. It performs no additional processing beyond the clustering, and a brute-force approach is required to search through any given cluster.
+
+##### IVFPQ
+Inverted File Product Quantization (IVFPQ) improves the vector storage cost by quantizing the data. It begins by clustering the data points, but before storing the vectors, it quantizes the data. Each vector in a cluster is divided into sub-vectors, and each sub-vector is encoded into bits using product quantization.
+
+At search time, the query vector is first associated with a cluster. Then, it is quantified, and the encoded sub-vectors are compared with the encoded data within the cluster. This storage method has various benefits, such as it saves on space, and since we’re comparing smaller vectors, it improves vector search times.
```

Comment:
```suggestion
At search time, the query vector is first associated with a cluster. Then, it is quantized, and the encoded sub-vectors are compared with the encoded data within the cluster. This storage method has various benefits, such as it saves on space, and since we’re comparing smaller vectors, it improves vector search times.
```

Reply:
maybe I am wrong here, but is  that what you meant?

if not, please clarify